include/hlslib/common/OpenCL.h

```diff
@@ -255,6 +255,28 @@ cl_mem_flags BankToFlag(MemoryBank memoryBank, bool failIfUnspecified,
   return 0;
 }
 
+MemoryBank StorageTypeToMemoryBank(StorageType storage, int bank) {
+  if (storage != StorageType::DDR) {
+    ThrowRuntimeError("Only DDR can be converted to Memorybank");
+  }
+  if (bank < 0 || bank > 3) {
+    ThrowRuntimeError("To large bank to convert to Memorybank");
+  }
+  switch (bank) {
+  case 0:
+    return MemoryBank::bank0;
+  case 1:
+    return MemoryBank::bank1;
+  case 2:
+    return MemoryBank::bank2;
+  case 3:
+    return MemoryBank::bank3;
+  default:
+    ThrowRuntimeError("Unsupported bank identifier.");
+  }
+  return MemoryBank::unspecified; // unreachable, but suppresses warnings
```

Comment:
```suggestion
```

Is this still necessary after you added the `default` value?

Reply:
g++ still issues a warning if not present "control reaches end of non void function". Technically it will compile and work without it too

Reply:
Ok, only if -wall is active, so I guess I will remove it